src/DaemonExec.ts

```diff
@@ -0,0 +1,244 @@
+import fs = require('fs')
+//import path = require('path')
+import Docker = require('dockerode')
+//needs docker to already be running
+const docker = new Docker()
+
+const listImages = async () => {
+	try {
+		const images = await docker.listImages({})
+		return images
+	} catch (err) {
+		console.log(err)
+		return []
+	}
+}
+
+const listContainers = async () => {
+	try {
+		const containers = await docker.listContainers()
+		return containers
+	} catch (err) {
+		console.log(err)
+		return []
+	}
+}
+
+const importImage = async (name: string) => {
+	try {
+		const stream = await docker.pull(name)
+		return stream
+	} catch (err) {
+		console.log(err)
+	}
+}
+
+//volumePairs should be of the form [[hostSrc,containerDest]]
+const createContainer = async (
+	image: string,
+	command: string[],
+	inputDir: string,
+	stdinFile: string,
+	outputDir: string,
+	stdoutFile: string,
+	errorDir: string,
+	stderrFile: string,
+	volumePairs: [string, string][],
+) => {
+	try {
+		const volumeJson: any = {}
+		for (const volumePair of volumePairs) {
+			volumeJson[`${volumePair[1]}`] = {}
+		}
+		const volumeArray = volumePairs.map((el) => el[0] + ':' + el[1])
+		const container = await docker.createContainer({
+			Image: image,
+			Cmd: command,
+			Volumes: volumeJson,
+			HostConfig: {
+				Binds: volumeArray,
+			},
+			AttachStdin: true,
+			AttachStdout: true,
+			AttachStderr: true,
+			Tty: false,
+			OpenStdin: true,
+			StdinOnce: true,
+		})
+		const stream = await container.attach({
+			hijack: true,
+			stream: true,
+			stdin: true,
+			stdout: true,
+			stderr: true,
+		})
+		let stdoutStream: any = process.stdout
+		if (outputDir != '' && stdoutFile != '') {
+			await fs.mkdir(outputDir, { recursive: true }, (err) => {
```

Comment:
Creating directories and writing files is distracting from the main point of this library. Why not just return the `Stream`s instead of taking dirs/files as arguments? That way the caller can do anything with them, such as forward them to the client directly through a socket. They can still log them to disk if they need that.